klogr.go

```diff
@@ -34,8 +39,13 @@ func NewKlogr() Logger {
 type klogger struct {
 	level     int
 	callDepth int
-	prefix    string
-	values    []interface{}
+
+	// havePrefix is true if the first entry in values is the special
+	// nameKey key/value. Such an entry gets added and later updated in
+	// WithName.
+	havePrefix bool
```

Comment:
Would it be better to name this as `hasPrefix`  ?

Reply:
Depends: I was thinking of it as "we have a prefix", but "logger has a prefix" also makes sense.

"has" seems to more popular in k/k/, although both occur - I'll switch to "has".